### PR TITLE
Add early_stop option, use print instead of summary.

### PR DESCRIPTION
**Pull request by SunQpark**
1. `early_stop` is added as suggested in #32. This feature can be set in the `early_stop` option in `config.json`. 

2. BaseModel class now override `__str__` function, instead of defining additional `summary` method to make it printed with number of params. This change replace `model.summary()` with `print(model)` in the `train.py`, making it does not crash when user does not inherit `BaseModel` and use `nn.Module`.

3. Changing config file layout on `monitor` option. Previously monitor mode and metric is set with two distinct options like `"monitor_mode = "min", monitor_metric = "val_loss"`. The same option for current layout is `"monitor" = "min val_loss"` now.

### CODE REVIEW

**Comment by SunQpark** [comment]
Corrected that line, thank you @victoresque.
